src/providers/DevicesProvider/AudioInputProvider.tsx

```diff
@@ -42,15 +43,28 @@ const AudioInputProvider: React.FC<Props> = ({
   const selectedInputRef = useRef(selectedAudioInputDevice);
   selectedInputRef.current = selectedAudioInputDevice;
 
-  const replaceDevice = async (device: string): Promise<AudioInputDevice> => {
-    if (onDeviceReplacement) {
-      return onDeviceReplacement(
-        device,
-        meetingManager.selectedAudioInputDevice
-      );
+  const replaceDevice = useCallback(
+    async (device: string): Promise<AudioInputDevice> => {
+      if (onDeviceReplacement) {
+        return onDeviceReplacement(
+          device,
+          meetingManager.selectedAudioInputDevice
+        );
+      }
+      return device;
+    },
+    [onDeviceReplacement, meetingManager.selectedAudioInputDevice]
+  );
+
+  const shouldAutoSelectAudioInputDevice = useCallback((): boolean => {
+    if (
+      meetingManager.getDeviceLabels() === DeviceLabels.Audio ||
+      meetingManager.getDeviceLabels() === DeviceLabels.AudioAndVideo
```

Comment:
Is it possible to use the `selectedAudioInputDevice` instead of the device labels? If the `selectedAudioInputDevice` is undefined, then we don't want to select the audio device that was just plugged in. Whats the difference between this and using device labels exactly?

Reply:
DeviceLabel is in our current way to track device permission. By setting it to None it would mean to not trigger any audio or video permission popup.

Reply:
"selectedAudioInputDevice being undefined is a result of builders explicitly setting DeviceLabels.None"

If this is true, then whats the difference between using `selectedAudioInputDevice` vs DeviceLabel.None? Shouldnt it be the same thing?

Reply:
DeviceLabel is a public facing knob to control whether to list and or select devices. For example: As part of meeting leave, we reset the `selectedAudioInputDevice` to `undefined`. Now the `DeviceLabels` will still be `DeviceLabels.AudioAndVideo` and only overridden when done through `join` or `invokeDeviceProvider` but the `selectedAudioInputDevice` for a new join will be `undefined` to start with and it will depend on `DeviceLabels` passed to next `join` to set `selectedAudioInputDevice` to correct device.

Thus, `selectedAudioInputDevice` is actually dependent on `DeviceLabels` been used. In v4, we will anyhow move away from the public fields on `MeetingManager` as those should actually be private (as applicable but selected devices fields all will be private).